cherryontech/jupiter-jumpers
# Update forgot pw styles, and add progress bars

## Description

1. Adds the updating styling to the new forgot password feature, and related components.
2. Adds a new progress tracker component to the different screens in the process
3. Adds a new confirmation screen after changing the password (used to redirect immediately to the login screen).

## Related Issue

[_Provide a link to the related issue in Jira._](https://cherryontech-jupiter-jumpers.atlassian.net/browse/SCRUM-42?atlOrigin=eyJpIjoiNzEyOTM4ZDdkNGM1NDIxZWJlMDI5NDg1NTJiZjA0ZWMiLCJwIjoiaiJ9)

## Related Story Card

**As a user**

- I want to reset my password if I forget it
- So that I can regain access to my account

## Type of Changes

Styling update, and feature addition

## Acceptance Criteria

- [X] - Given I am on the login screen
- [X] - When I select the "Forgot Password?" link
- [X] - Then I am taken through a guided process to reset my password securely.


## Update Screenshots

_Provide screenshots of the view before and after the changes._

### Before

### After

<img width="1333" alt="Screenshot 2024-05-09 at 8 20 37 PM" src="https://github.com/cherryontech/jupiter-jumpers/assets/47403135/260bcb64-dd40-422c-94b1-584d5db46a76">
<img width="1329" alt="Screenshot 2024-05-09 at 8 20 51 PM" src="https://github.com/cherryontech/jupiter-jumpers/assets/47403135/c7e63b30-6086-4efb-9fb1-69a4b4a0586a">
<img width="1326" alt="Screenshot 2024-05-09 at 8 21 36 PM" src="https://github.com/cherryontech/jupiter-jumpers/assets/47403135/2319acbd-68df-43ff-ba4b-7d1c2038aa6f">
<img width="1331" alt="Screenshot 2024-05-09 at 8 21 57 PM" src="https://github.com/cherryontech/jupiter-jumpers/assets/47403135/88f6037a-2b8a-4a3e-afe9-c7f61829a8d1">

## Testing Instructions

1. In the login view, select "Forgot password"
2. You should see the "Forgot Password?" view for users to enter their email (with progress bar at 1)
3. Enter a previously-registered email.
4. Ensure you see the "Check your email" screen (with progress bar at 2)
5. Ensure you receive the email to reset your password.
6. Click the link in the email, and ensure you see the screen to reset your password (with progress bar at 3)
7. When you click "Reset Password", ensure you are taken to the "Password successfully changes!" screen.
8. Click "Log in" and ensure you are redirected to the login view.


**Self-Reminder (Gisele):** Change the Firebase URL back after merging.
